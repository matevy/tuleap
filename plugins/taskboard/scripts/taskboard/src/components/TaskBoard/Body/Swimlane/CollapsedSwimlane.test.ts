/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import CollapsedSwimlane from "./CollapsedSwimlane.vue";
import { createTaskboardLocalVue } from "../../../../helpers/local-vue-for-test";
import { createStoreMock } from "../../../../../../../../../src/www/scripts/vue-components/store-wrapper-jest";
import { ColumnDefinition, Swimlane } from "../../../../type";
import { RootState } from "../../../../store/type";

describe("CollapsedSwimlane", () => {
    it("displays a toggle icon and a card with minimal information", async () => {
        const $store = createStoreMock({
            state: {
                swimlane: {},
                column: {
                    columns: [] as ColumnDefinition[]
                }
            } as RootState
        });
        const wrapper = shallowMount(CollapsedSwimlane, {
            localVue: await createTaskboardLocalVue(),
            mocks: { $store },
            propsData: {
                swimlane: {
                    card: {
                        color: "fiesta-red"
                    }
                } as Swimlane
            }
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("expand the swimlane when user click on the toggle icon", async () => {
        const $store = createStoreMock({
            state: {
                swimlane: {},
                column: {
                    columns: [] as ColumnDefinition[]
                }
            } as RootState
        });
        const swimlane: Swimlane = {
            card: {
                color: "fiesta-red"
            }
        } as Swimlane;
        const wrapper = shallowMount(CollapsedSwimlane, {
            localVue: await createTaskboardLocalVue(),
            mocks: { $store },
            propsData: { swimlane }
        });
        wrapper.find(".taskboard-swimlane-toggle").trigger("click");
        expect($store.dispatch).toHaveBeenCalledWith("swimlane/expandSwimlane", swimlane);
    });
});
