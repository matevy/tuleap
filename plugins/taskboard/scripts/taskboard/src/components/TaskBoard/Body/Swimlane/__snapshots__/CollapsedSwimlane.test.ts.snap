// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsedSwimlane displays a toggle icon and a card with minimal information 1`] = `
<div
  class="taskboard-swimlane taskboard-swimlane-collapsed"
>
  <swimlane-header-stub
    swimlane="[object Object]"
  >
    <i
      class="fa fa-plus-square taskboard-swimlane-toggle tlp-swatch-fiesta-red"
      role="button"
      title="Expand"
    />
     
    <div
      class="taskboard-card taskboard-card-collapsed taskboard-card-fiesta-red"
    >
      <div
        class="taskboard-card-content"
      >
        <card-xref-label-stub
          card="[object Object]"
          class="taskboard-card-xref-label-collapsed"
        />
      </div>
    </div>
  </swimlane-header-stub>
</div>
`;
